### ISSUE TITLE
SS4: MigrateFileTask throws validation exceptions for error pages

### ISSUE THREAD

**Issue by madmatt** [silverstripe/silverstripe-framework#5560]
When migrating site assets, it seems the default `File.allowed_extensions` doesn't include .html, so running the `MigrateFilesTask` fails with a `ERROR [Emergency]: Uncaught ValidationException: Extension is not allowed (valid: <list of extensions>)` error message.

Adding the following to a mysite/_config.php fixes the issue:
```yml
File:
  allowed_extensions:
    - 'html'
```

**Comment by madmatt** [comment]
It seems like we should allow HTML as an allowed extension, I can't see any particular issues with that? I'm not sure how this has slipped through before, it's not in 3.x either, but we're now validating the filename in more places it seems.

**Comment by micmania1** [comment]
I think this enables .html file uploads on all upload fields by default, which is bad :p

**Comment by madmatt** [comment]
Yeah, I guess? It seems like there should be some rule - either HTML can be uploaded by default, or the error page assets shouldn't be treated as File objects. The issue stems from the file extension validation only being applied during an upload in 3.x (not on `File->write()`), but now as part of 4.x it gets checked during filesystem operations (e.g. when moving files between public/protected adapters).

As it stands right now, as far as I can tell any site that has the error-404.html or error-500.html (which I'd imagine is most sites) will fail during upgrade, and you'll need to add a workaround (which for most people is likely to be exactly this).

OTOH, I knew it would raise some concerns, so that's why I haven't raised it directly as a pull request ;)

On the subject of bad file extensions to allow, we currently allow `.dmg` files (Mac executables), `.jar` files, `.js` files, as well as all the major archive formats under the sun (`tar`, `gz` etc.) which can easily be used to hide the real files... I think I might raise a separate pull request to fix that.

**Comment by tractorcow** [comment]
Error pages shouldn't be treated as File objects.

Additionally, if a file in 3.x has an allowed_extension that isn't allowed in 4.x then I would probably expect the importer to log this as a warning, and skip attempting to import that file.

Site developers can then read these messages and manually whitelist extensions as necessary.
